db/active_replicator_pull.go

```diff
@@ -158,10 +154,6 @@ func (apr *ActivePullReplicator) _subChanges(collectionIdx *int, since string) e
 func (apr *ActivePullReplicator) Complete() {
 	base.TracefCtx(apr.ctx, base.KeyReplicate, "ActivePullReplicator.Complete()")
 	apr.lock.Lock()
-	if apr == nil {
-		apr.lock.Unlock()
-		return
-	}
```

Comment:
Similarly, I'm not sure of the original motivation for this guard? Maybe being overly defensive?

Reply:
I think this was overly defensive programming, and is not hit currently.